### PR TITLE
Endpointer register

### LINKED ISSUE #6: [feature] add bp-init to register the service on the endpointer
create bp-init dll to register the service on the endpointer, and init all the stuff that are required for each service

### PR DESCRIPTION
**Pull request by joni7777**
resolves #6

### CODE REVIEW

File: Bp.ApiRunner/Runner.cs [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-                .UseKestrel((builderContext, options) =>
-                {
-                    options.Configure(builderContext.Configuration.GetSection("Kestrel"));
-                })
-                .UseSerilog(SerilogInit.ConfigureLogger);
-
-            builder.Build().Run();
-        }
-    }
+	public class Runner
```
Comment (Meir017) [inline Bp.ApiRunner/Runner.cs:8]: weird spacing in file

File: Bp.EndPointer/Bp.EndPointer.csproj [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>netcoreapp2.2</TargetFramework>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <Reference Include="Microsoft.AspNetCore.Hosting, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60">
```
Comment (Meir017) [inline Bp.EndPointer/Bp.EndPointer.csproj]: shouldn't these be `PackageReference`?

File: Bp.EndPointer/EndPointer.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+				if (!respone.IsSuccessStatusCode)
+				{
+					_logger?.LogWarning("Failed to register to the endpointer", respone.ReasonPhrase);
+				}
+				else
+				{
+					_logger?.LogInformation("Registered successfully to the endpointer", respone.ReasonPhrase);
+				}
+			}
+			catch (HttpRequestException e) when (e.Message == "Connection refused")
```
Comment (Meir017) [inline Bp.EndPointer/EndPointer.cs]: there should be another catch clause for the Task Cancelled exception

File: Bp.EndPointer/RegisterEndpointerHostedService.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		{
+			return RegisterToEndPointer();
+		}
+
+		private async Task RegisterToEndPointer()
+		{
+			await _endPointer.Register(_serverAddress);
+		}
+
+		private string ParseServerAddress(string address)
```
Comment (Meir017) [inline Bp.EndPointer/RegisterEndpointerHostedService.cs:42]: add a description explaining what the input address looks like

File: Bp.Logging/Sinks/Mattermost/BpMattermostSinkExtensions.cs [hunk 5]
```diff
@@ -9,9 +9,11 @@ public static class BpMattermostSinkExtensions
     {
         public static LoggerConfiguration BpMattermostSink(
             this LoggerSinkConfiguration loggerConfiguration,
-            BpMattermostInfo bpMattermostInfo, HttpClient httpClient = null, LogEventLevel restrictedToMinimumLevel = LogEventLevel.Verbose)
+            BpMattermostInfo bpMattermostInfo, HttpClient httpClient = null,
+            LogEventLevel restrictedToMinimumLevel = LogEventLevel.Verbose)
```
Comment (Meir017) [inline Bp.Logging/Sinks/Mattermost/BpMattermostSinkExtensions.cs]: I think the default should be `Error`

File: Bp.Logging/Sinks/Splunk/BpSplunkInfo.cs [hunk 6]
```diff
@@ -13,13 +13,15 @@ public class BpSplunkInfo
...
-            !String.IsNullOrEmpty(Protocol) &&!String.IsNullOrEmpty(Host) &&
+            !String.IsNullOrEmpty(Protocol) && !String.IsNullOrEmpty(Host) &&
             !String.IsNullOrEmpty(Port) && !String.IsNullOrEmpty(Username) &&
             !String.IsNullOrEmpty(Password) && !String.IsNullOrEmpty(Index) &&
             !String.IsNullOrEmpty(SourceType);
 
         public const string SPLUNK_MISSING_CONFIGURATION_MESSAGE = "Splunk missing required configuration";
 
-        public BpSplunkInfo(){}
+        public BpSplunkInfo()
```
Comment (Meir017) [inline Bp.Logging/Sinks/Splunk/BpSplunkInfo.cs]: no need for ctor

**Review by Meir017 — CHANGES_REQUESTED** [review]

File: Bp.ApiRunner/Runner.cs (hunk 1 above)
Comment (joni7777) [reply Bp.ApiRunner/Runner.cs:8]: Looks kinda fine to me, can you be more specific about the problem?

Comment (Meir017) [reply Bp.ApiRunner/Runner.cs:8]: tabs vs spaces...
never mind

File: Bp.EndPointer/EndPointer.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
```
Comment (Meir017) [inline Bp.EndPointer/EndPointer.cs]: maybe use `catch () when ()` ?
something like this:
```cs
catch (Exception e) 
when ((e is HttpRequestException && e.Message == "Connection refused")
    || e is TaskCanceledException)
{
    ....
    throw;
}
```

**Review by Meir017 — APPROVED** [review]

**Pull request merged**